### ISSUE TITLE
Make google_analytics more generic

### ISSUE THREAD

**Issue by robodude666** [mkdocs/mkdocs#661]
`google_analytics` is currently the only official config option for adding analytics to your docs, unless you want to use extras and deal with `config.extra....` which isn't very pretty.

Certainly would be solved with #206 plugin support, however, the `google_analytics` option could be made more generic:

```yaml
analytics:
   type: google
   url: mkdocs.org
   tracking: UA-1234567-8
```

By default `google` would be supported and used to validate `url` and `tracking` are present. If a non-builtin type is detected, such as `piwik`, no validation is done and data is provided to jinja2 as `{{ analytics['key'] }}`.

Could possibly be expanded to an analytics plugin offering different types with their own generated html.

**Comment by waylan** [comment]
This appears to be related to #597, which was closed in favor of #607. As this issue is related to the config settings more than the theme, it is not clear to me if this is a duplicate or not. However, if #607 was completed, this request would be unnecessary as you could override the necessary bits of the theme and simply add the appropriate values to `config.extra`.

**Comment by d0ugal** [comment]
Yeah, let's focus on #607 and try and fix this more generally. There is no way I want to try supporting tons of different analytics options. I'd like to remove the Google analytics support for 1.0 if we can and delegate this either to more powerful theme customisation or extensions.
